iquip/apps/explorer.py

```diff
@@ -72,7 +95,8 @@ def _addFile(self, path: str, parent: Union[QTreeWidget, QTreeWidgetItem]):
             if experimentFile.endswith("/"):
                 experimentFileItem = QTreeWidgetItem(parent)
                 experimentFileItem.setText(0, experimentFile[:-1])
```

Comment:
Since you are now run this codes in another thread, you have to be aware of the GUI event loop in the main thread.
If you update the GUI (e.g., `setText()`) in a thread other than the main thread, it could crash.
For safety, you should separate the _working_ part and the GUI part, and the GUI part must run on the main thread.
In order to accomplish this, you may use signal-slot mechanism or [`invokeMethod()`](https://doc.qt.io/qt-6/qmetaobject.html#invokeMethod-7). You might need `QThread` for these feature (I'm not sure!).
Or you could find a nice way to use `threading.Thread` instead.

Reply:
I found out a few solutions of setting a callback method, but they are a little complicated.
So I decide to replace it to `QThread`.